src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -188,56 +189,54 @@ static BlockItRange mapValueIdItPairToBlockRange(const ValueIdIt& idRangeBegin,
   // `BlockIdIt`s. This is why we have to divide by two in the following after
   // we calculated the respective `ValueId` index with
   // std::distance(idRangeBegin, idIt).
-  return BlockItRange{
-      BlockIt{blockRangeBegin + std::distance(idRangeBegin, beginIdIt) / 2},
-      std::min(blockRange.end(),
-               BlockIt{blockRangeBegin +
-                       (std::distance(idRangeBegin, endIdIt) + 1) / 2})};
+  auto blockOffsetBegin = std::distance(idRangeBegin, beginIdIt) / 2;
+  auto blockOffsetEnd = (std::distance(idRangeBegin, endIdIt) + 1) / 2;
+  return {blockRangeBegin + blockOffsetBegin,
+          std::min(blockRange.end(), blockRangeBegin + blockOffsetEnd)};
 }
 
 //______________________________________________________________________________
 // Map the complement on the given `ValueIdItPairs`s to their corresponding
-// `BlockItRange`s.
+// `BlockSubrange`s.
 // The actual mapping is implemented by `mapValueIdItPairToBlockRange`.
-RelevantBlockItRanges mapValueIdItRangesToBlockItRangesComplemented(
+BlockSubranges mapValueIdItRangesToBlockItRangesComplemented(
     const std::vector<ValueIdItPair>& relevantIdRanges,
-    const ValueIdItRange& idRange, BlockSpan blockRange) {
+    const ValueIdSubrange& idRange, BlockSpan blockRange) {
   auto idRangeBegin = idRange.begin();
   // Vector containing the `BlockRange` mapped indices.
-  RelevantBlockItRanges blockRanges;
+  BlockSubranges blockRanges;
   blockRanges.reserve(relevantIdRanges.size());
-
-  ValueIdIt lastIdRangeIt = idRange.begin();
+  auto addRange =
+      std::bind_front(mergeBlockRangeWithRanges, std::ref(blockRanges));
```

Comment:
```suggestion
      absl::bind_front(mergeBlockRangeWithRanges, std::ref(blockRanges));
```
(include `absl/functional/bind_front.h`)